feat(impl):[TRI-742] callback url implementation

ds-ext-kmassalski opened the pull request:
- sending JobProcessingFinishedEvent after marking job as completed/error
- listens for JobProcessingFinishedEvent  and calling callbackUrl if is valid
- logging response to console

ds-jkreutzfeld reviewed (changes requested):
The changes look good, but while testing the feature, I found that I could never reach the log output "Callback url pinged, received http status" (JobProcessingEventListener#66) when using unavailable services (http://example.org)
The noErrorRestTemplate logs an error instead.

ds-ext-kmassalski commented:
> The changes look good, but while testing the feature, I found that I could never reach the log output "Callback url pinged, received http status" (JobProcessingEventListener#66) when using unavailable services (http://example.org) The noErrorRestTemplate logs an error instead.

- added catching of connection timeout exception,
- added @URL validator for POST job registration request
